lib/dal/tests/integration_test/internal/new_engine/frame.rs

```diff
@@ -106,3 +109,769 @@ async fn convert_component_to_frame_and_attach(ctx: &mut DalContext) {
         fallout_parent_node_id.expect("no parent node id for fallout component")
     );
 }
+
+#[test]
+async fn multiple_frames_with_complex_connections_no_nesting(ctx: &mut DalContext) {
```

Comment:
This test runs through multiple scenarios with frames (and no nesting) in the new engine.